### ISSUE TITLE
Why split all feedstocks into one repo each?

### ISSUE THREAD

**Issue by nils-werner** [conda-forge/feedstocks#3]
I think the approach of having one repo per feedstock is not a good design choice because It scales really badly

 - A repo with 100 directories is more manageable than 100 repos (discovery, issues, merge requests, dependencies)
 - Changes to the CI scripts need to be merged 100 times
 - The correct way of submitting new recipes is really opaque (MR to staged-recipes, but a new repo for what should be a merge)

Now I see why you'd want to have separate repos: A rebuild of all recipes on each commit would not be feasible. However I think  a "smart CI script" could solve that:

 1. Lint all recipes
 2. Only build recipes that were actually changed.

**Comment by pelson** [comment]
Hi @nils-werner. We discussed this topic in a bit of detail at https://github.com/conda-forge/conda-forge.github.io/issues/25.

Essentially, we have been down the road of one repository with lots of recipes. I wrote much of the infrastructure to do that for repositories such as https://github.com/ioos/conda-recipes and https://github.com/SciTools/conda-recipes-scitools amongst others. They worked really well (the IOOS channel is exceptionally well curated, and has served as a massive asset to the earth sciences for over a year), but the approach simply doesn't scale to the level that we need. We want to package many repositories, we want a place where people can go to discuss in detail about specific building techniques, we want the ability to control *who* has access to *what*, we want to assert that *every* time a repository changes the recipe continues to build.

We can certainly do a better job of documenting these lessons we've learnt the hard way - that is certainly on my todo list over the next few weeks and months.

Thanks,
